### PR TITLE
fix(ebsr): set only one correct answer when controls are switched from checkbox to radio button PD-262

### PR DESCRIPTION
**Pull request by CarlaCostea**
https://illuminate.atlassian.net/browse/PD-262

### CODE REVIEW

File: packages/ebsr/configure/src/main.jsx [hunk 1]
```diff
@@ -26,12 +26,42 @@ export class Main extends React.Component {
     onConfigurationChanged: PropTypes.func
   };
 
+  onModelChanged = (model, key) => {
+    const { onModelChanged } = this.props;
+
+    switch (key) {
```
Comment (andreeapescar) [inline packages/ebsr/configure/src/main.jsx:32]: No need for "switch" here, you only have one different case to treat.

File: packages/ebsr/configure/src/main.jsx [hunk 2]
```diff
@@ -26,12 +26,42 @@ export class Main extends React.Component {
...
+          model.partB.choices = model.partB.choices.map(c => {
+            if (correctFound) {
+              c.correct = false;
+              return c;
+            }
+
+            if (c.correct) {
+              correctFound = true;
+            }
+            return c;
```
Comment (andreeapescar) [inline packages/ebsr/configure/src/main.jsx:48]: I think you can use only one "if" here:

if (c.correct && correctFound) {
c.correct = false;
correctFound = true;
}
return c;

Please correct me if this doesn't what you're expecting.

**Review by andreeapescar — CHANGES_REQUESTED** [review]

File: packages/ebsr/configure/src/main.jsx (hunk 1 above)
Comment (CarlaCostea) [reply packages/ebsr/configure/src/main.jsx:32]: ok

File: packages/ebsr/configure/src/main.jsx (hunk 2 above)
Comment (CarlaCostea) [reply packages/ebsr/configure/src/main.jsx:48]: 'correctFound' becomes true when first correct answer is found. then all 'c.correct' are made false

File: packages/ebsr/configure/src/main.jsx [hunk 3]
```diff
@@ -26,12 +26,41 @@ export class Main extends React.Component {
...
   };
 
+  onModelChanged = (model, key) => {
+    const { onModelChanged } = this.props;
+
+    if (key !== 'partB.choiceMode') {
+      return onModelChanged(model);
+    };
+
+    let value = model.partB.choiceMode;
```
Comment (andreeapescar) [inline packages/ebsr/configure/src/main.jsx:36]: "value" can be const (it doesn't get updated)

**Review by andreeapescar — APPROVED** [review]

File: packages/ebsr/configure/src/main.jsx (hunk 3 above)
Comment (CarlaCostea) [reply packages/ebsr/configure/src/main.jsx]: sure

**Review by andreeapescar — APPROVED** [review]

**Pull request merged**